src/Feature/LanguageSupport/Common.re

```diff
@@ -5,4 +5,8 @@ type outmsg =
   | OpenFile({
       filePath: string,
       location: option(Location.t),
+    })
+  | PreviewFile({
+      filePath: string,
+      location: option(Location.t),
```

Comment:
It looks like this isn't actually used by any of the LanguageSupport features - seems like we could simplify and remove this in `Feature_LanguageSupport` (and remove the handler in `Features.re`).

Reply:
@bryphe Are you sure that we won't need that in future?
For example in vscode `Go To Definition` also opens in `preview` mode.
We could use it in future

Reply:
I'd rather see it added when it's hooked up - so the usage is clearer. Until we're actually using it, it would just add a maintenance burden to `Features.re` and `Feature_LanguageSupport` (in other words - it's something else we have to refactor!). 

In addition, until it is actually used, we might not have the interface totally right - once we have a use case, that connection will be more crystallized and we can make sure we have it right. We also don't have a way to test this functionality w/o some connection.

So to me, this seems like a case of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), at least until we have a usage. Perhaps a subsequent change could wire up 'Go to Definition' to preview, and add this plumbing as part of it.